### PR TITLE
VR-185: adding demoMode env variable.

### PR DESCRIPTION
**Pull request by n3op2**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

## PR Type

Please delete options that are irrelevant.

- [x] Feature

## Linked tickets

https://digicatapult.atlassian.net/browse/VR-185

## High level description

Adding environment variable to the helm-chart (DEMO_MODE)

> pod details
```sh
(feature/vr-185) $ kubectl describe pod  test-veritable-ui-cbf878cb4-kr7zn
Name:             test-veritable-ui-cbf878cb4-kr7zn
Namespace:        default
Priority:         0
Service Account:  test-veritable-ui
Node:             kind-control-plane/172.19.0.2
Start Time:       Tue, 27 Aug 2024 10:58:21 +0100
Labels:           app.kubernetes.io/component=veritable-ui
                  app.kubernetes.io/instance=test
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=veritable-ui
                  app.kubernetes.io/version=v0.8.29
                  helm.sh/chart=veritable-ui-1.0.5
                  pod-template-hash=cbf878cb4
Annotations:      <none>
Status:           Running
IP:               10.244.0.6
IPs:
  IP:           10.244.0.6
Controlled By:  ReplicaSet/test-veritable-ui-cbf878cb4
Init Containers:
  database-migrate:
    Container ID:  containerd://43bef2522aaca4a4684569af92b22dfcff9356afcc9b1292c5c7907962f6adfe
    Image:         docker.io/digicatapult/veritable-ui:v0.8.29
    Image ID:      docker.io/digicatapult/veritable-ui@sha256:34325f2466c9c0305f7da514e876ee4daeb1fbc34ca088516419e78b3b2353dd
    Port:          <none>
    Host Port:     <none>
    Command:
      npx
      knex
    Args:
      migrate:latest
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 27 Aug 2024 11:00:21 +0100
      Finished:     Tue, 27 Aug 2024 11:00:21 +0100
    Ready:          True
    Restart Count:  3
    Environment:
      NODE_ENV:     production
      DB_HOST:      test-vui-postgresql
      DB_PORT:      5432
      DB_NAME:      veritable-ui
      DB_USERNAME:  veritable-ui
      DB_PASSWORD:  <set to the key 'password' in secret 'test-vui-postgresql'>  Optional: false
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-sms9j (ro)
  issueance-policies:
    Container ID:  containerd://729a83cd939e4c379fda36c1907d75c2e839310618814a4ad57d2010aab0b9eb
    Image:         docker.io/digicatapult/veritable-ui:v0.8.29
    Image ID:      docker.io/digicatapult/veritable-ui@sha256:34325f2466c9c0305f7da514e876ee4daeb1fbc34ca088516419e78b3b2353dd
    Port:          <none>
    Host Port:     <none>
    Command:
      npm
      run
      init
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 27 Aug 2024 11:01:08 +0100
      Finished:     Tue, 27 Aug 2024 11:01:10 +0100
    Ready:          True
    Restart Count:  3
    Environment:
      CLOUDAGENT_ADMIN_ORIGIN:   http://test-cloudagent-admin:3000
      LOG_LEVEL:                 debug
      ISSUANCE_DID_POLICY:       EXISTING_OR_NEW
      ISSUANCE_SCHEMA_POLICY:    EXISTING_OR_NEW
      ISSUANCE_CRED_DEF_POLICY:  EXISTING_OR_NEW
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-sms9j (ro)
Containers:
  veritable-ui:
    Container ID:   containerd://be1c690b00b1fd00370ba54c987fe8059d723eed1f8467da2806baad0681b0f0
    Image:          docker.io/digicatapult/veritable-ui:v0.8.29
    Image ID:       docker.io/digicatapult/veritable-ui@sha256:34325f2466c9c0305f7da514e876ee4daeb1fbc34ca088516419e78b3b2353dd
    Port:           3000/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Tue, 27 Aug 2024 11:01:11 +0100
    Ready:          True
    Restart Count:  0
    Liveness:       http-get http://:http/connection delay=10s timeout=5s period=10s #success=1 #failure=3
    Readiness:      http-get http://:http/connection delay=5s timeout=5s period=10s #success=1 #failure=5
    Startup:        http-get http://:http/connection delay=30s timeout=5s period=10s #success=1 #failure=10
    Environment:
      COOKIE_SESSION_KEYS:             <set to the key 'secret' in secret 'test-veritable-ui-cookie-session-keys'>  Optional: false
      PUBLIC_URL:                      http://localhost:3080
      API_SWAGGER_BG_COLOR:            #fafafa
      API_SWAGGER_TITLE:               Veritable
      API_SWAGGER_HEADING:             Veritable
      IDP_CLIENT_ID:                   veritable-ui
      IDP_CLIENT_ID:                   veritable-ui
      IDP_PUBLIC_URL_PREFIX:           http://localhost:3080/realms/veritable/protocol/openid-connect
      IDP_INTERNAL_URL_PREFIX:         http://localhost:3080/realms/veritable/protocol/openid-connect
      IDP_AUTH_PATH:                   /auth
      IDP_TOKEN_PATH:                  /token
      IDP_JWKS_PATH:                   /certs
      COMPANY_HOUSE_API_URL:           https://api.company-information.service.gov.uk
      COMPANY_PROFILE_API_KEY:         <set to the key 'secret' in secret 'test-veritable-ui-company-house-api-key'>  Optional: false
      PORT:                            3000
      LOG_LEVEL:                       debug
      DB_HOST:                         test-vui-postgresql
      DB_PORT:                         5432
      DB_NAME:                         veritable-ui
      DB_USERNAME:                     veritable-ui
      DB_PASSWORD:                     <set to the key 'password' in secret 'test-vui-postgresql'>  Optional: false
      EMAIL_TRANSPORT:                 STREAM
      EMAIL_FROM_ADDRESS:              hello@veritable.com
      EMAIL_ADMIN_ADDRESS:             admin@veritable.com
      CLOUDAGENT_ADMIN_WS_ORIGIN:      ws://test-cloudagent-admin:3000
      CLOUDAGENT_ADMIN_ORIGIN:         http://test-cloudagent-admin:3000
      INVITATION_FROM_COMPANY_NUMBER:  00000000
      INVITATION_PIN_SECRET:           <set to the key 'pin' in secret 'test-veritable-ui-invitation-pin'>  Optional: false
      INVITATION_PIN_ATTEMPT_LIMIT:    3
      ISSUANCE_DID_POLICY:             FIND_EXISTING
      ISSUANCE_SCHEMA_POLICY:          FIND_EXISTING
      ISSUANCE_CRED_DEF_POLICY:        FIND_EXISTING
      >>>>>>>>>>> DEMO_MODE:                       false <<<<<<<<<<<
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-sms9j (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   True 
  Initialized                 True 
  Ready                       True 
  ContainersReady             True 
  PodScheduled                True 
Volumes:
  kube-api-access-sms9j:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason        Age                    From               Message
  ----     ------        ----                   ----               -------
  Normal   Scheduled     5m30s                  default-scheduler  Successfully assigned default/test-veritable-ui-cbf878cb4-kr7zn to kind-control-plane
  Normal   Pulling       5m29s                  kubelet            Pulling image "docker.io/digicatapult/veritable-ui:v0.8.29"
  Normal   Pulled        4m8s                   kubelet            Successfully pulled image "docker.io/digicatapult/veritable-ui:v0.8.29" in 26.447s (1m20.821s including waiting)
  Warning  BackOff       3m42s (x4 over 4m6s)   kubelet            Back-off restarting failed container database-migrate in pod test-veritable-ui-cbf878cb4-kr7zn_default(345022d7-89b5-45b3-a81b-da3d18b3a6f4)
  Normal   Created       3m30s (x4 over 4m8s)   kubelet            Created container database-migrate
  Normal   Started       3m30s (x4 over 4m8s)   kubelet            Started container database-migrate
  Normal   Pulled        3m30s (x3 over 4m7s)   kubelet            Container image "docker.io/digicatapult/veritable-ui:v0.8.29" already present on machine
  Normal   Created       3m28s (x2 over 3m29s)  kubelet            Created container issueance-policies
  Normal   Started       3m28s (x2 over 3m29s)  kubelet            Started container issueance-policies
  Warning  BackOff       3m27s                  kubelet            Back-off restarting failed container issueance-policies in pod test-veritable-ui-cbf878cb4-kr7zn_default(345022d7-89b5-45b3-a81b-da3d18b3a6f4)
  Normal   Pulled        3m12s (x3 over 3m29s)  kubelet            Container image "docker.io/digicatapult/veritable-ui:v0.8.29" already present on machine
  Warning  ProbeWarning  19s (x11 over 119s)    kubelet            Liveness probe warning: Probe terminated redirects, Response body: Found. Redirecting to http://localhost:3080/auth/login?path=%2Fconnection
```

### CODE REVIEW

File: charts/veritable-ui/templates/deployment.yaml [hunk 1]
```diff
@@ -252,6 +252,8 @@ spec:
               value: {{ .Values.schemaPolicy | quote }}
             - name: ISSUANCE_CRED_DEF_POLICY
               value: {{ .Values.credentialPolicy | quote }}
+            - name: DEMO_MODE 
+              value: {{ include "veritable-ui.demoMode" . | quote }}
```
Comment (mattdean-digicatapult) [inline charts/veritable-ui/templates/deployment.yaml]: Can't this just be `value: {{ .Values.demoMode | quote }}`?

Comment (mattdean-digicatapult) [reply charts/veritable-ui/templates/deployment.yaml]: if not where is the template you're including?

Comment (n3op2) [reply charts/veritable-ui/templates/deployment.yaml]: it meant to be, should have updated with https://github.com/digicatapult/helm-charts/pull/114/commits/560cb19456befbf31ddffbfd544b9b43c8ae079d commit

**Review by mattdean-digicatapult — APPROVED** [review]
Looks good assuming tests pass

File: charts/veritable-ui/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ annotations:
   category: veritable
   licenses: Apache-2.0
 apiVersion: v2
-appVersion: v0.8.13
```
Comment (rmlearney-digicatapult) [inline charts/veritable-ui/Chart.yaml:5]: Does renovate not catch this?

**Review by Ellenn-A — APPROVED** [review]
looks good

File: charts/veritable-ui/values.yaml [hunk 3]
```diff
@@ -707,7 +709,7 @@ externalDatabase:
 ## @param cloudagent.postgresql.nameOverride Name for the veritable-cloudagent PostgreSQL dependency
 ##
 cloudagent:
-  enabled: false
+  enabled: true
```
Comment (dblane-digicatapult) [inline charts/veritable-ui/values.yaml:712]: revert

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**